app/assets/javascripts/autocomplete_checkboxes.js

```diff
@@ -0,0 +1,90 @@
+function hitEnter(event, element, formType, inputType) {
+  if (event.which === 13) {
+    enterOnAutocomplete(element, formType, inputType);
```

Comment:
okay, so outlining my steps.
1. I see the default input field text message  `Press ENTER to create a category` 
2. I hit enter a few times, but there is no action. (A little confused)
3. I enter a character (but it also removes the default help message), then it makes sense.

It feels like this wants to be a tooltip. I've used `tooltipster` to get that, so we might be able to open a separate improvement down the road.
https://github.com/Amunds/tooltipster-rails
http://iamceege.github.io/tooltipster/

For now the code looks good, so we can choose to defer to a new issue.